jena-extras/jena-serviceenhancer/pom.xml

```diff
@@ -0,0 +1,185 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Please clean the code.

I get 52 warnings in Eclipse.
  Missing overrides, unused imports, bad javadoc (various kinds), bad try-with-resources, unnecessary casts.

mvn (with java11) javadoc:javadoc has 86 lines of javadoc warnings

There are many commented code fragments.

There multiple adjacent blank lines.

Reply:
`mvn clean install` and `mvn javadoc:javadoc` succeeds without any warnings at least for me.

Reply:
I am still get 36 warning from Eclipse:

* Javadoc : `@return` without text after it.
* try-with-resource
* Missing `@Override`
* Unnecessary casts

Eclispe or IntelliJ will pick these up.

When people start working with the Jena codebase, it's nice to give them a reasonable clean view.

As Release Manager, it is also helpful to have a clean view so I can see when new things arise.

```
Javadoc: Description expected after @return	BatchQueryRewriter.java	line 129
Javadoc: Description expected after @return	IteratorFactoryWithBuffer.java	line 96
Javadoc: Description expected after @return	Ref.java	line 73
Javadoc: Description expected after @return	ServiceOpts.java	line 118
Javadoc: Description expected after @return	Slice.java	line 57
Javadoc: Description expected after @return	Slice.java	line 90
Javadoc: Description expected after @return	VarScopeUtils.java	line 105
Resource 'disposable' should be managed by try-with-resource	AsyncClaimingCacheImplGuava.java	line 378	Java Problem
Resource 'reclaim' should be managed by try-with-resource	AsyncClaimingCacheImplGuava.java	line 385
Resource 'ref' should be managed by try-with-resource	AsyncClaimingCacheImplGuava.java	line 376
The method claim(K) of type AsyncClaimingCacheImplGuava<K,V> should be tagged with @Override since it actually overrides a superinterface method	AsyncClaimingCacheImplGuava.java	line 162
The method close() of type SliceAccessor<A> should be tagged with @Override since it actually overrides a superinterface method	SliceAccessor.java	line 108
The method createSubIterator(boolean) of type IteratorFactoryWithBuffer<T,I>.SubIteratorImpl should be tagged with @Override since it actually overrides a superinterface method	IteratorFactoryWithBuffer.java	line 135
The method exec(OpService) of type OpServiceExecutorImpl should be tagged with @Override since it actually overrides a superinterface method	OpServiceExecutorImpl.java	line 51
The method getArrayOps() of type BufferLike<A> should be tagged with @Override since it actually overrides a superinterface method	BufferLike.java	line 33
The method getDelegate() of type RefDelegateBase<T,R> should be tagged with @Override since it actually overrides a superinterface method	RefDelegateBase.java	line 31
The method getFailedRanges() of type SliceMetaDataImpl should be tagged with @Override since it actually overrides a superinterface method	SliceMetaDataImpl.java	line 70
The method getItems() of type BatchImpl<K,T> should be tagged with @Override since it actually overrides a superinterface method	BatchImpl.java	line 76
The method getLoadedRanges() of type SliceMetaDataImpl should be tagged with @Override since it actually overrides a superinterface method	SliceMetaDataImpl.java	line 62
The method getMaximumKnownSize() of type SliceMetaDataImpl should be tagged with @Override since it actually overrides a superinterface method	SliceMetaDataImpl.java	line 78
The method getMinimumKnownSize() of type SliceMetaDataImpl should be tagged with @Override since it actually overrides a superinterface method	SliceMetaDataImpl.java	line 74
The method getNextValidIndex() of type BatchImpl<K,T> should be tagged with @Override since it actually overrides a superinterface method	BatchImpl.java	line 68
The method getRanges() of type RangeBufferImpl<A> should be tagged with @Override since it actually overrides a superinterface method	RangeBufferImpl.java	line 90
The method isEmpty() of type BatchImpl<K,T> should be tagged with @Override since it actually overrides a superinterface method	BatchImpl.java	line 80
The method newSliceAccessor() of type SliceWithPages<T> should be tagged with @Override since it actually overrides a superinterface method	SliceWithPages.java	line 31
The method put(K, T) of type BatchImpl<K,T> should be tagged with @Override since it actually overrides a superinterface method	BatchImpl.java	line 58
The method size() of type BatchImpl<K,T> should be tagged with @Override since it actually overrides a superinterface method	BatchImpl.java	line 84
The method start() of type ServiceEnhancerInit should be tagged with @Override since it actually overrides a superinterface method	ServiceEnhancerInit.java	line 58
The method unsafeRead(A, int, long, int) of type SliceAccessorImpl<A> should be tagged with @Override since it actually overrides a superinterface method	SliceAccessorImpl.java	line 286
Unnecessary cast from A to A	BufferOverArray.java	line 50
Unnecessary cast from int to long	PageUtils.java	line 44
Unnecessary cast from K to K	AsyncClaimingCacheImplGuava.java	line 307
Unnecessary cast from long to long	RangeBufferImpl.java	line 83
Unnecessary cast from Ref to Ref<T>	RefImpl.java	line 162
Unnecessary cast from V to V	AsyncClaimingCacheImplGuava.java	line 307
Unnecessary cast from V to V	AsyncClaimingCacheImplGuava.java	line 324
```

Reply:
Hm, do you know if it's possible to generate these warnings also with a cli `mvn` invocation?
At least adding `-Pdev` or `-Dmaven.compiler.showWarnings` does not generate them.

I can warnings with IntelliJ analyze code though - will go through them.

Reply:
These are IDE warnings (IntelliJ does, or should, have the same.)

They aren't more than general housekeeping, except for the `@return` which causes a "return" section with no content.

Reply:
In my Eclipse 2022-06 I had to manually enable the mentioned warnings - they now seem to be on ignore by default (at least I clicked on restore defaults and they remained on ignore) - I am pretty sure they were enabled in the past though.

Anyway, all the mentioned ones plus some additional ones should now be fixed and I squashed - can you confirm?

I am currently re-deploying to our test bed to make sure that my changes don't break the fuseki setup.

Reply:
There are a few "info" warning but nothing that can't be sorted up automatically which I will do. 

They are warning that suppressed warnings (for "resource") are hidden by other settings which I guess is  whether resource leaks for potential leaks is enabled as well as definite resource leaks. I have them off because where Jena predates try-resource, there significant numbers of warnings and need design changes to fix.

So good to merge!

Phew!

Reply:
I noticed those potential resource leaks warnings - and I wonder if there is any meaningful design pattern to make those go away:
Essentially these methods allocate resources which are used in the set up of the query iterators. Because of the streaming approach they can only be closed once the iterator is closed. Wrapping them with try-with-resources does not make sense there.